security-framework/src/passwords.rs

```diff
@@ -204,14 +208,47 @@ fn internet_password_query(
     query
 }
 
+bitflags::bitflags! {
+    /// The option flags used to configure the evaluation of a `SecAccessControl`.
+    pub struct AccessControlOptions: CFOptionFlags {
+        /** Constraint to access an item with either biometry or passcode. */
+        const USER_PRESENCE = kSecAccessControlUserPresence;
+        /** Constraint to access an item with Touch ID for any enrolled fingers, or Face ID. */
+        const BIOMETRY_ANY = kSecAccessControlBiometryAny;
+        /** Constraint to access an item with Touch ID for currently enrolled fingers, or from Face ID with the currently enrolled user. */
+        const BIOMETRY_CURRENT_SET = kSecAccessControlBiometryCurrentSet;
+        /** Constraint to access an item with a passcode. */
+        const DEVICE_PASSCODE = kSecAccessControlDevicePasscode;
+        /** Constraint to access an item with a watch. */
+        const WATCH = kSecAccessControlWatch;
+        /** Indicates that at least one constraint must be satisfied. */
+        const OR = kSecAccessControlOr;
+        /** Indicates that all constraints must be satisfied. */
+        const AND = kSecAccessControlAnd;
+        /** Enable a private key to be used in signing a block of data or verifying a signed block. */
+        const PRIVATE_KEY_USAGE = kSecAccessControlPrivateKeyUsage;
+        /** Option to use an application-provided password for data encryption key generation. */
+        const APPLICATION_PASSWORD = kSecAccessControlApplicationPassword;
+    }
+}
+
 // This starts by trying to create the password with the given query params.
 // If the creation attempt reveals that one exists, its password is updated.
-fn set_password_internal(query: &mut Vec<(CFString, CFType)>, password: &[u8]) -> Result<()> {
+fn set_password_internal(query: &mut Vec<(CFString, CFType)>, password: &[u8], access_control_options: Option<AccessControlOptions>) -> Result<()> {
     let query_len = query.len();
     query.push((
         unsafe { CFString::wrap_under_get_rule(kSecValueData) },
         CFData::from_buffer(password).as_CFType(),
     ));
+
+    if let Some(options) = access_control_options {
+        let error = ptr::null_mut();
```

Comment:
this doesn't look right. If you're trying to capture the error, this should be defined in an outer scope and use `&mut error` later. Otherwise just use `ptr::null_mut()` without assigning it to a variable first.